### ISSUE TITLE
Revert changes removing scaleFactor from loss functions

### ISSUE THREAD

**Issue by mbojrab** [mbojrab/playbox#66]
The supervised algorithms have consistent loss and associated learningRates. However the greedy unsupervised algorithms need learningRates associated with each layer depending on its size. This is cumbersome to train appropriately/consistently as networks sizes change. 

Instead use the older scheme to scale based on neurons/batchSize for unsupervised algorithms. This allows similar learningRates even as the network topology wildly changes.

**Comment by mbojrab** [comment]
closed by #67
